purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/BasePurchasesIntegrationTest.kt

```diff
@@ -0,0 +1,145 @@
+package com.revenuecat.purchases
+
+import android.content.Context
+import android.preference.PreferenceManager
+import androidx.test.ext.junit.rules.activityScenarioRule
+import com.revenuecat.purchases.common.BillingAbstract
+import com.revenuecat.purchases.models.StoreTransaction
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import org.assertj.core.api.Assertions
+import org.junit.BeforeClass
+import org.junit.Rule
+import java.net.URL
+import java.util.Date
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import kotlin.time.Duration.Companion.seconds
+
+open class BasePurchasesIntegrationTest {
+
+    companion object {
+        @BeforeClass
+        @JvmStatic
+        fun setupClass() {
+            if (!canRunIntegrationTests()) {
+                error("You need to set required constants in Constants.kt")
+            }
+        }
+
+        private fun canRunIntegrationTests() = Constants.apiKey != "REVENUECAT_API_KEY" &&
+            Constants.googlePurchaseToken != "GOOGLE_PURCHASE_TOKEN" &&
+            Constants.productIdToPurchase != "PRODUCT_ID_TO_PURCHASE"
+    }
+
+    @get:Rule
+    var activityScenarioRule = activityScenarioRule<MainActivity>()
+
+    protected val testTimeout = 7.seconds
+    protected val currentTimestamp = Date().time
+    protected val testUserId = "android-integration-test-$currentTimestamp"
+    protected val proxyUrl = Constants.proxyUrl.takeIf { it != "NO_PROXY_URL" }
+
+    protected lateinit var mockBillingAbstract: BillingAbstract
+
+    protected var latestPurchasesUpdatedListener: BillingAbstract.PurchasesUpdatedListener? = null
+    protected var latestStateListener: BillingAbstract.StateListener? = null
+
+    protected val entitlementsToVerify = Constants.activeEntitlementIdsToVerify
+        .split(",")
+        .map { it.trim() }
+        .filter { it.isNotEmpty() }
+
+    // region helpers
+
+    protected fun setupTest(
```

Comment:
Nit: `set up` should be a verb here which is 2 words
```suggestion
    protected fun setUpTest(
```